src/backend/bs_worker

```diff
@@ -3128,7 +3133,13 @@ sub dobuild {
     unshift @configpath, "path=$projid/$repoid" unless @configpath && $configpath[0] eq "path=$projid/$repoid";
   }
   my $server = $buildinfo->{'srcserver'} || $srcserver;
-  my $config = BSRPC::rpc("$server/getconfig", undef, "project=$projid", "repository=$repoid", @configpath);
+  my $config;
+  if (defined $buildinfo->{'config'}) {
+    $config = $buildinfo->{'config'};
+  } else {
+    $config = BSRPC::rpc("$server/getconfig", undef, "project=$projid", "repository=$repoid", @configpath);
+  }
+  $buildinfo->{'outbuildinfo'}->{'config'} = $config; # the bs_worker code sets $outbuildinfo->{'config'} to the build config $config
```

Comment:
this is bs_worker code, you readbuildenv method instead?

Reply:
I removed the comment. It created more confusion than clarity.